e2e-tests/src/test/scala/pl/touk/nussknacker/BatchDataGenerationSpec.scala

```diff
@@ -95,9 +95,17 @@ class BatchDataGenerationSpec
       .when()
       .request()
       .basicAuthAdmin()
-      .jsonBody(toScenarioGraph(simpleBatchTableScenarioLiveMode).asJson.spaces2)
+      .jsonBody(
+        s"""{
+           | "testData": {
+           |   "type": "WITH_GENERATED_DATA",
+           |   "numberOfSamples": 1
+           | },
+           | "scenarioGraph": ${toScenarioGraph(simpleBatchTableScenarioLiveMode).asJson.spaces2}
+           |}""".stripMargin
+      )
       .post(
-        s"$designerServiceUrl/api/processManagement/generateAndTest/$scenarioName/1"
+        s"$designerServiceUrl/api/scenarioTest/$scenarioName/test"
```

Comment:
As we discussed, let's add some tests at the lower level, using `ScenarioBuilder`, checking the result of the map with transition results. Cases that IMO would be nice to cover are:
- variable
- split
- for-each
- collect
- union
- fragment

Let's discuss if you find some case problematic to cover. Take a look at `ConfigCreatorWithCollectingListener` usages where we do similar things.